connector/connect/common/src/main/protobuf/spark/connect/types.proto

```diff
@@ -184,3 +184,15 @@ message DataType {
     DataType sql_type = 5;
   }
 }
+
+enum StorageLevel {
```

Comment:
the problem is that the definitions are different between PySpark and SQL:

PySpark: `MEMORY_AND_DISK = useDisk=True, useMemory=True, useOffHeap=False, deserialized=False`
SQL: `MEMORY_AND_DISK = _useDisk=true, _useMemory=true, _useOffHeap=false, _deserialized=true`

also `MEMORY_AND_DISK_DESER ` was dedicated for PySpark, it is equivalent to `MEMORY_AND_DISK` in Scala side.

cc @cloud-fan @HyukjinKwon @amaliujia 

also please follow [the guide](https://protobuf.dev/programming-guides/style/#enums) to define an `enum`, it should be like:
```
enum StorageLevel {
    STORAGE_LEVEL_UNSPECIFIED = 0;
    STORAGE_LEVEL_ DISK_ONLY = 1;
   ...
```

Reply:
@zhengruifeng thanks for review. 
I have removed enum and static mappings. 
Now, storage level is resolved based on user input. It allows Python users to use PySpark constants or customise if needed.  This the same logic how PySpark accepts and resolves storage level from user.